main.rb

```diff
@@ -0,0 +1,117 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      if instance_of(Hash) || instance_of?(Range)
+        new_hash = to_a
+        yield(new_hash[counter])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+  end
```

Comment:
#### my_each
- [ ] 1. #my_each returns an Enumerator if no block is given
- [ ] 2. #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- [ ] 3. #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- [ ] 4. #my_each_with_index returns an enumerator if no block is given
- [ ] 5. #my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- [ ] 6. #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self